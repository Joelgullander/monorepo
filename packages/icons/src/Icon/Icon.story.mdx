import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { action } from '@storybook/addon-actions';

import { Icon } from './Icon';

<Meta title='Components|Icons' component={Icon} decorators={[withKnobs]} />

# Icons

<Preview>
    <Story name="plus">
        <Icon name="plus" />   
    </Story>
    <Story name="comments">
        <Icon name="comments" />   
    </Story>
    <Story name="down small arrow">
        <Icon name="downSmallArrow" />   
    </Story>
     <Story name="right small arrow">
        <Icon name="rightSmallArrow" />   
    </Story>
    <Story name="close">
        <Icon name="close" />   
    </Story>
    <Story name="search">
        <Icon name="search" />   
    </Story>
    <Story name="leftArrow">
        <Icon name="leftArrow" />   
    </Story>
    <Story name="alertCircle">
        <Icon name="alertCircle" />   
    </Story>
</Preview>


<Preview>
    <Story name="folder">
        <Icon name="folder" />     
    </Story>
    <Story name="openfolder">
        <Icon name="openFolder" />     
    </Story>
    <Story name="circleChecked">
        <Icon name="circleChecked" />     
    </Story>
    <Story name="circleClose">
        <Icon name="circleClose" />     
    </Story>
    <Story name="circleQuestion">
        <Icon name="circleQuestion" />     
    </Story>
    <Story name="globe">
        <Icon name="globe" />     
    </Story>
    <Story name="rightArrow">
        <Icon name="rightArrow" />   
    </Story>
</Preview>

<Preview>
    <Story name="forestlyCranB">
        <Icon
            width="103px"
            height="67px"
            name="forestlyCranB"
        />     
    </Story>
    <Story name="forestlyCranW">
        <Icon
            width="103px"
            height="67px"
            name="forestlyCranW"
        />     
    </Story>
</Preview>

<Preview>
    <Story name="hookliftSkiploaderB">
        <Icon
            width="103px"
            height="67px"
            name="hookliftSkiploaderB"
        />     
    </Story>
    <Story name="hookliftSkiploaderW">
        <Icon
            width="103px"
            height="67px"
            name="hookliftSkiploaderW"
        />     
    </Story>
</Preview>

<Preview>
    <Story name="loaderCraneB">
        <Icon
            width="103px"
            height="67px"
            name="loaderCraneB"
        />     
    </Story>
    <Story name="loaderCraneW">
        <Icon
            width="103px"
            height="67px"
            name="loaderCraneW"
        />     
    </Story>
</Preview>

<Preview>
    <Story name="recyclingCraneB">
        <Icon
            width="103px"
            height="67px"
            name="recyclingCraneB"
        />     
    </Story>
    <Story name="recyclingCraneW">
        <Icon
            width="103px"
            height="67px"
            name="recyclingCraneW"
        />     
    </Story>
</Preview>

<Preview>
    <Story name="tailliftB">
        <Icon
            width="103px"
            height="67px"
            name="tailliftB"
        />     
    </Story>
    <Story name="tailliftW">
        <Icon
            width="103px"
            height="67px"
            name="tailliftW"
        />     
    </Story>
</Preview>

<Preview>
    <Story name="truckMountedB">
        <Icon
            width="103px"
            height="67px"
            name="truckMountedB"
        />     
    </Story>
    <Story name="truckMountedW">
        <Icon
            width="103px"
            height="67px"
            name="truckMountedW"
        />     
    </Story>
</Preview>

## Props

<Props of={Icon} />
