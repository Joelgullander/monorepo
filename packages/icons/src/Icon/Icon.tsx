import React from 'react';
import {
  CommentsIcon,
  PlusIcon,
  DownSmallArrowIcon,
  CloseIcon,
  SearchIcon,
  LeftArrowIcon,
  RightArrowIcon,
  AlertCircleIcon,
  FolderIcon,
  OpenFolderIcon,
  CircleCheckedIcon,
  CircleCloseIcon,
  GlobeIcon,
  HiabIcon,
  CircleQuestionIcon,
  RightSmallArrowIcon,
  ForestlyCranB,
  ForestlyCranW,
  HookliftSkiploaderB,
  HookliftSkiploaderW,
  LoaderCraneB,
  LoaderCraneW,
  RecyclingCraneB,
  RecyclingCraneW,
  TailliftB,
  TailliftW,
  TruckMountedB,
  TruckMountedW
} from '../assets';

export enum Name {
  PLUS = 'plus',
  COMMENTS = 'comments',
  DOWN_SMALL_ARRROW = 'downSmallArrow',
  CLOSE = 'close',
  SEARCH = 'search',
  LEFT_ARROW = 'leftArrow',
  RIGHT_ARROW = 'rightArrow',
  ALERT_CIRCLE = 'alertCircle',
  FOLDER = 'folder',
  OPEN_FOLDER = 'openFolder',
  CIRCLE_CHECKED = 'circleChecked',
  CIRCLE_CLOSE = 'circleClose',
  GLOBE = 'globe',
  HIAB = 'hiab',
  CIRCLE_QUESTION = 'circleQuestion',
  RIGHT_SMALL_ARROW = 'rightSmallArrow',
  FORESTLY_CRAN_B = 'forestlyCranB',
  FORESTLY_CRAN_W = 'forestlyCranW',
  HOOKLIFT_SKIPLOADER_B = 'hookliftSkiploaderB',
  HOOKLIFT_SKIPLOADER_W = 'hookliftSkiploaderW',
  LOADER_CRANE_B = 'loaderCraneB',
  LOADER_CRANE_W = 'loaderCraneW',
  RECYCLING_CRANE_B = 'recyclingCraneB',
  RECYCLING_CRANE_W = 'recyclingCraneW',
  TAILLIFT_B = 'tailliftB',
  TAILLIFT_W = 'tailliftW',
  TRUCK_MOUNTED_B = 'truckMountedB',
  TRUCK_MOUNTED_W = 'truckMountedW',
};


export interface IIcon {
  name: Name,
  width?: string,
  height?: string,
  color?: string,
};

const IconsWrapper: React.FC<IIcon> = ({ name, ...props }) => {
  switch (name) {
    case Name.PLUS:
      return <PlusIcon {...props} />;
    case Name.COMMENTS:
      return <CommentsIcon {...props} />;
    case Name.DOWN_SMALL_ARRROW:
      return <DownSmallArrowIcon {...props} />;
    case Name.CLOSE:
      return <CloseIcon {...props} />;
    case Name.SEARCH:
      return <SearchIcon {...props} />;
    case Name.LEFT_ARROW:
      return <LeftArrowIcon {...props} />;
    case Name.RIGHT_ARROW:
      return <RightArrowIcon {...props} />;
    case Name.ALERT_CIRCLE:
      return <AlertCircleIcon {...props} />
    case Name.FOLDER:
      return <FolderIcon {...props} />
    case Name.OPEN_FOLDER:
      return <OpenFolderIcon {...props} />
    case Name.CIRCLE_CHECKED:
      return <CircleCheckedIcon {...props} />
    case Name.CIRCLE_CLOSE:
      return <CircleCloseIcon {...props} />
    case Name.GLOBE:
      return <GlobeIcon {...props} />
    case Name.HIAB:
      return <HiabIcon {...props} />
    case Name.CIRCLE_QUESTION:
      return <CircleQuestionIcon {...props} />
    case Name.RIGHT_SMALL_ARROW:
      return <RightSmallArrowIcon {...props} />
    case Name.FORESTLY_CRAN_B:
      return <ForestlyCranB {...props} />
    case Name.FORESTLY_CRAN_W:
      return <ForestlyCranW {...props} />
    case Name.HOOKLIFT_SKIPLOADER_B:
      return <HookliftSkiploaderB {...props} />
    case Name.HOOKLIFT_SKIPLOADER_W:
      return <HookliftSkiploaderW {...props} />
    case Name.LOADER_CRANE_B:
      return <LoaderCraneB {...props} />
    case Name.LOADER_CRANE_W:
      return <LoaderCraneW {...props} />
    case Name.RECYCLING_CRANE_B:
      return <RecyclingCraneB {...props} />
    case Name.RECYCLING_CRANE_W:
      return <RecyclingCraneW {...props} />
    case Name.TAILLIFT_B:
      return <TailliftB {...props} />
    case Name.TAILLIFT_W:
      return <TailliftW {...props} />
    case Name.TRUCK_MOUNTED_B:
      return <TruckMountedB {...props} />
    case Name.TRUCK_MOUNTED_W:
      return <TruckMountedW {...props} />
    default:
      return <PlusIcon {...props} />
  }
};

export const Icon: React.FC<IIcon> = (props) => {
  return (
    <IconsWrapper {...props} />
  );
};

Icon.defaultProps = {
  width: '25px',
  height: '25px',
  color: '#000',
};
