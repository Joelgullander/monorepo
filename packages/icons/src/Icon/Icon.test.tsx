import * as React from 'react';

import { shallow } from 'enzyme';
import { Icon, Name } from './Icon';

describe('<Icon />', () => {
  test('snapshot', () => {
    const icon = shallow(<Icon name={Name.PLUS} />);
    
    expect(icon).toMatchSnapshot();
  });

  test('Icon prop name is passing', () => {
    const icon = shallow(<Icon name={Name.PLUS} />);
    
    expect(icon.props().name).toEqual(Name.PLUS);
  });

  test('Icon prop width is passing', () => {
    const icon = shallow(<Icon width="50" name={Name.PLUS} />);
    
    expect(icon.props().width).toEqual("50");
  });
});
