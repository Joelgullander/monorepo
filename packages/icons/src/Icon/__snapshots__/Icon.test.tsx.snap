// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> snapshot 1`] = `
<IconsWrapper
  color="#000"
  height="25px"
  name="plus"
  width="25px"
/>
`;
